### PR TITLE
RR-579 - Prisoner movement model and skeleton service class

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR models the prisoner movement data we receive from the `prison-api` into a simplified set of data classes. The mapping between these and the `prison-api` classes, plus the subsequent inclusion into the prisoner `Timeline` domain will be done in subsequent PRs (I've added TODOs where relevant).

This PR also reverts the temporary code I added to the `JpaTimelinePersistenceAdapter` to test the connection to the `prison-api`.

### CODE REVIEW

File: domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/service/TimelineService.kt [hunk 1]
```diff
@@ -34,6 +35,11 @@ class TimelineService(
    * Returns the [Timeline] for the prisoner identified by their prison number. Otherwise, throws
    * [TimelineNotFoundException] if it cannot be found.
    */
-  fun getTimelineForPrisoner(prisonNumber: String): Timeline =
-    persistenceAdapter.getTimelineForPrisoner(prisonNumber) ?: throw TimelineNotFoundException(prisonNumber)
+  fun getTimelineForPrisoner(prisonNumber: String): Timeline {
+    val prisonerTimeline =
+      persistenceAdapter.getTimelineForPrisoner(prisonNumber) ?: throw TimelineNotFoundException(prisonNumber)
+
+    prisonerTimeline.addEvents(prisonTimelineService.getPrisonTimelineEvents(prisonNumber))
```
Comment (mattwills-moj-digital) [inline domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/service/TimelineService.kt:42]: We simply add additional `TimelineEvents` here, which will ended up being sorted by date, along with the other existing events.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaTimelinePersistenceAdapterTest.kt [hunk 2]
```diff
@@ -28,6 +27,7 @@ import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.aVal
 @ExtendWith(MockitoExtension::class)
 class JpaTimelinePersistenceAdapterTest {
```
Comment (mattwills-moj-digital) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaTimelinePersistenceAdapterTest.kt:28]: This class has simply been reverted back to its previous state, before I tested the connection with the prison-api.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaTimelinePersistenceAdapter.kt [hunk 3]
```diff
@@ -22,9 +19,6 @@ class JpaTimelinePersistenceAdapter(
   private val timelineRepository: TimelineRepository,
   private val timelineMapper: TimelineEntityMapper,
   private val timelineEventMapper: TimelineEventEntityMapper,
-  // TODO RR-566 - call prisonApiClient via service class
-  private val prisonApiClient: PrisonApiClient,
-  @Value("\${featureToggles.call-prison-api-enabled}") private val callPrisonApiEnabled: Boolean,
 ) : TimelinePersistenceAdapter {
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaTimelinePersistenceAdapter.kt:22]: This class has simply been reverted back to its previous state, before I tested the connection with the prison-api.

File: domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/service/TimelineService.kt [hunk 4]
```diff
@@ -34,6 +36,13 @@ class TimelineService(
...
    * [TimelineNotFoundException] if it cannot be found.
    */
-  fun getTimelineForPrisoner(prisonNumber: String): Timeline =
-    persistenceAdapter.getTimelineForPrisoner(prisonNumber) ?: throw TimelineNotFoundException(prisonNumber)
+  fun getTimelineForPrisoner(prisonNumber: String): Timeline {
+    val prisonerTimeline =
+      persistenceAdapter.getTimelineForPrisoner(prisonNumber) ?: throw TimelineNotFoundException(prisonNumber)
+
+    if (callPrisonApiEnabled) {
+      prisonerTimeline.addEvents(prisonTimelineService.getPrisonTimelineEvents(prisonNumber))
```
Comment (mattwills-moj-digital) [inline domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/service/TimelineService.kt:44]: We simply add additional TimelineEvents here, which will ended up being sorted by date, along with the other existing events.

File: domain/timeline/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/service/TimelineServiceTest.kt [hunk 5]
```diff
@@ -2,27 +2,38 @@ package uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.ser
...
   private lateinit var persistenceAdapter: TimelinePersistenceAdapter
 
+  @Mock
+  private lateinit var prisonTimelineService: PrisonTimelineService
+
+  private val callPrisonApiEnabled = true
+
+  @BeforeEach
+  fun setup() {
+    service = TimelineService(persistenceAdapter, prisonTimelineService, callPrisonApiEnabled)
```
Comment (mattwills-moj-digital) [inline domain/timeline/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/service/TimelineServiceTest.kt:34]: Once we remove the `callPrisonApiEnabled` feature toggle, I'll revert this test back to use `@InjectMocks`.

**Review by nathanrussell-moj-digital — APPROVED** [review]
A fine body of work 👍

**Pull request merged**